sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/ErrorReceiver.java

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common;
+
+/**
+ * An {@code ErrorReceiver} is an object that can be used to report errors on transfers. When specified on transfer
+ * operations, the {@code reportError} method will be called whenever errors are encountered. The user may configure
```

Comment:
1. Is this still just for non-fatal errors?
2. Why is it generic? Are there multiple kinds of errors?

Reply:
I made it generic in case we want to reuse this for something else some day, but I might be being over eager.

Reply:
For now at least, this is for fatal and non fatal errors after our discussion with the service team.